### ISSUE TITLE
train

### ISSUE THREAD

**Issue by BigdateAB** [mzzcdf/Thermal3DGS#9]
I don't know if it's the initialization of the model parameters or the learning rate, but the results of my experiments were so poor that I didn't change the default parameters.

![Image](https://github.com/user-attachments/assets/3acef2ac-52ab-4c8c-94aa-7fe9fd6f869d)

**Comment by BigdateAB** [comment]
The training metrics stayed the same, oddly enough

**Comment by wpd-mlb** [comment]
I also encountered the same problem, can the author interpret it?

**Comment by BIn-999** [comment]
I encountered the same problem: the rendered image was completely white. I printed out the intermediate results and found that the ATF stats gradually decreased from -0.004 to -60, -600, and -600,000, exponentially decreasing. The output of torch.exp(stats) , d_rgb, dropped from 1 to 0, eventually becoming all 0.
The raw render also decreased in sync, eventually becoming all 0.
In other words, the values generated by the ATF module were diverging and becoming negative, causing the entire render to be multiplied by 0.


[Debug] ATF stats min/max = -0.004/0.106 [01/08 17:01:16]
[Debug] d_rgb min/max = 0.996/1.111 [01/08 17:01:16]
[Debug] raw render min/max = 0.3459/0.4981 [01/08 17:01:16]
[Debug] ATF stats min/max = -0.346/-0.004 [01/08 17:01:16]
[Debug] d_rgb min/max = 0.707/0.996 [01/08 17:01:16]
[Debug] raw render min/max = 0.2542/0.4580 [01/08 17:01:16]
[Debug] ATF stats min/max = -1.661/-0.010 [01/08 17:01:16]
[Debug] d_rgb min/max = 0.190/0.990 [01/08 17:01:16]
[Debug] raw render min/max = 0.2904/0.4457 [01/08 17:01:16]
[Debug] ATF stats min/max = -4.445/-0.021 [01/08 17:01:16]
[Debug] d_rgb min/max = 0.012/0.980 [01/08 17:01:16]
[Debug] raw render min/max = 0.2519/0.3919 [01/08 17:01:16]
[Debug] ATF stats min/max = -9.273/-0.040 [01/08 17:01:16]
[Debug] d_rgb min/max = 0.000/0.961 [01/08 17:01:16]
[Debug] raw render min/max = 0.2861/0.4497 [01/08 17:01:16]
[Debug] ATF stats min/max = -17.480/-0.078 [01/08 17:01:16]
[Debug] d_rgb min/max = 0.000/0.925 [01/08 17:01:16]
[Debug] raw render min/max = 0.2114/0.3510 [01/08 17:01:16]
[Debug] ATF stats min/max = -31.813/-0.128 [01/08 17:01:16]
[Debug] d_rgb min/max = 0.000/0.880 [01/08 17:01:16]
[Debug] raw render min/max = 0.1951/0.3530 [01/08 17:01:16]
[Debug] ATF stats min/max = -61.032/-0.224 [01/08 17:01:16]
[Debug] d_rgb min/max = 0.000/0.799 [01/08 17:01:16]
[Debug] raw render min/max = 0.1683/0.2796 [01/08 17:01:16]
[Debug] ATF stats min/max = -121.331/-0.506 [01/08 17:01:16]
[Debug] d_rgb min/max = 0.000/0.603 [01/08 17:01:16]
[Debug] raw render min/max = 0.0599/0.1202 [01/08 17:01:16]
[Debug] ATF stats min/max = -253.065/-1.022 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.360 [01/08 17:01:17]
[Debug] raw render min/max = 0.0183/0.0484 [01/08 17:01:17]
Training progress:   0%|                                                                                    | 10/30000 [00:00<31:26, 15.89it/s, Loss=0.5543706][Debug] ATF stats min/max = -614.453/-2.395 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.091 [01/08 17:01:17]
[Debug] raw render min/max = 0.0012/0.0031 [01/08 17:01:17]
[Debug] ATF stats min/max = -1622.804/-6.664 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.001 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -3946.440/-13.603 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -8799.674/-27.706 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -18492.402/-67.634 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -34763.723/-114.763 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -61475.453/-197.835 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -101670.500/-354.638 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -158334.922/-463.422 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -233359.969/-753.932 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
Training progress:   0%|                                                                                    | 20/30000 [00:01<29:17, 17.06it/s, Loss=0.4522295][Debug] ATF stats min/max = -334767.438/-1084.253 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -455369.938/-1319.256 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -602792.562/-1873.789 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
[Debug] raw render min/max = 0.0000/0.0000 [01/08 17:01:17]
[Debug] ATF stats min/max = -771731.750/-2017.836 [01/08 17:01:17]
[Debug] d_rgb min/max = 0.000/0.000 [01/08 17:01:17]
